src/metabase/api/action.clj

```diff
@@ -84,14 +84,30 @@
       ;; so we return the most recently updated http action.
       (last (action/select-actions :type "http")))))
 
-(api/defendpoint-schema PUT "/:id"
+(def ^:private json-query-schema
+  [:and
+   string?
+   [:fn {:error/message "must be a valid json-query"
+         :description "must be a valid json-query"}
+    #(http-action/apply-json-query {} %)]])
+
+(def ^:private http-action-template
+  [:map {:closed true}
+   [:method                              [:enum "GET" "POST" "PUT" "DELETE" "PATCH"]]
+   [:url                                 [string? {:min 1}]]
+   [:body               {:optional true} [:maybe string?]]
+   [:headers            {:optional true} [:maybe string?]]
+   [:parameters         {:optional true} [:maybe [:sequential map?]]]
```

Comment:
a question: what is the different if we use `:*` instead of `:sequential` here?

Reply:
I think they're the same, but I think of `:*` as being part of the sequence regex schemas, so I wouldn't reach for it unless you're doing something that requires it. [regex schema ex.](https://malli.io/?value=%5B%3Anames%20%5B%22a%22%20%22b%22%5D%20%3Anums%20%5B1%202%203%5D%5D&schema=%5B%3Acat%20%5B%3A%3D%20%3Anames%5D%20%5B%3Aschema%20%5B%3A*%20string%3F%5D%5D%20%5B%3A%3D%20%3Anums%5D%20%5B%3Aschema%20%5B%3A*%20number%3F%5D%5D%5D)